File: evidence/generated-systems/angular/governance-on-angular/src/app/models/DataSubjectRequest.js

// Define collection and schema for DataSubjectRequest
export interface DataSubjectRequest {
    receivedDate:
	type : Date
    dueDate:
	type : Date
    requesterCountry:
	type : string
    Organization:
	type : Schema.Types.ObjectId
    ProcessingActivities:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataProcessingActivity' }]
    Records:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Record_' }]
    RequestType:
 	type : String
    Status:
 	type : String
#
    collection: 'dataSubjectRequests'
}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/datasubjectrequest.rb
class DataSubjectRequest < ApplicationRecord
  enum RequestType: [:Access, :Rectification, :Erasure, :Restriction, :Portability, :Objection, :AutomatedDecisioningReview]
  enum Status: [:Received, :InValidation, :InProgress, :OnHold, :Fulfilled, :Rejected]


  has_many :Organization, class_name: 'Organization'
  has_many :ProcessingActivities, class_name: 'DataProcessingActivity'
  has_many :Records, class_name: 'Record_'

end
